realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.android;
+
+import android.content.ContentResolver;
+import android.database.CharArrayBuffer;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.DataSetObservable;
+import android.database.DataSetObserver;
+import android.net.Uri;
+import android.os.Bundle;
+
+import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmResults;
+import io.realm.internal.ColumnType;
+import io.realm.internal.TableOrView;
+
+/**
+ * This classes exposes {@link io.realm.RealmResults} as a cursor.
+ * <p>
+ * It is possible to traverse links using dot notation to access data in referenced objects, ie.
+ * {@code cursor.getInt("foo.bar"} will return the integer from the {@code bar} field in the {@code foo} object
+ * referenced the class represented by the RealmResult.
+ * <p>
+ * Many Android framework classes require the presences of an "_id" field. Instead of adding such a field to your
+ * model class it is instead possible to use {@link #setIdColumn(String)}.
+ * <p>
+ * A RealmCursor has the same thread restrictions as RealmResults, so it is not possible to move a RealmCursor between
+ * threads.
+ * <p>
+ * TODO How to handle RealmList? getString("realmList[0].name)"?
+ * TODO How to iterate a RealmList?
+ */
+public class RealmCursor implements Cursor {
+
+    private static final String DEFAULT_ID_COLUMN = "_id";
+
+    private final Realm realm;
+    private TableOrView table;
+    private int rowIndex;
+    private boolean closed;
+    private long idColumnIndex = -1; // Column index for the "_id" field. -1 means it hasn't been set
+    private final DataSetObservable dataSetObservable = new DataSetObservable();
+    private RealmChangeListener changeListener = new RealmChangeListener() {
+        @Override
+        public void onChange() {
+            if (dataSetObservable != null) {
+                dataSetObservable.notifyChanged(); // TODO Should be replaced with something more fine grained.
+            }
+        }
+    };
+
+    /**
+     * Exposes a RealmResults object as a cursor. Use {@link RealmResults#getCursor()} instead of this
+     * constructor
+     *
+     * @param table Table view representing the query results.
+     */
+    public RealmCursor(Realm realm, TableOrView table) {
```

Comment:
Because they are in different packages. RealmResults are in ```io.realm``` while this is located in ```io.realm.android```.  #1093 is about the same.